pkg/auth/oauth2/auth_openshift.go

```diff
@@ -130,51 +151,21 @@ func (o *openShiftAuth) getOIDCDiscoveryInternal(ctx context.Context) (*oidcDisc
 	return metadata, nil
 }
 
-func (o *openShiftAuth) login(w http.ResponseWriter, _ *http.Request, token *oauth2.Token) (*sessions.LoginState, error) {
+func (o *openShiftAuth) login(w http.ResponseWriter, r *http.Request, token *oauth2.Token) (*sessions.LoginState, error) {
 	if token.AccessToken == "" {
 		return nil, fmt.Errorf("token response did not contain an access token %#v", token)
 	}
-	ls := sessions.NewRawLoginState(token.AccessToken)
-
-	expiresIn := (time.Hour * 24).Seconds()
-	if !token.Expiry.IsZero() {
-		expiresIn = token.Expiry.Sub(time.Now()).Seconds()
-	}
 
-	// NOTE: In Tectonic, we previously had issues with tokens being bigger than
-	// cookies can handle. Since OpenShift doesn't store groups in the token, the
-	// token can't grow arbitrarily big, so we assume it will always fit in a cookie
-	// value.
-	//
-	// NOTE: in the future we'll have to avoid the use of cookies. This should likely switch to frontend
-	// only logic using the OAuth2 implicit flow.
-	// https://tools.ietf.org/html/rfc6749#section-4.2
-	cookie := http.Cookie{
-		Name:     sessions.OpenshiftAccessTokenCookieName,
-		Value:    ls.AccessToken(),
-		MaxAge:   int(expiresIn),
-		HttpOnly: true,
-		Path:     o.cookiePath,
-		Secure:   o.secureCookies,
-		SameSite: http.SameSiteStrictMode,
+	ls, err := o.sessions.AddSession(w, r, nil, token)
```

Comment:
Shouldnt we be verifying the token here, similarly to what the oidc_auth is doing ?
```suggestion
	ls, err := o.sessions.AddSession(w, r, o.verify, token)
```

Reply:
From my understanding of it, the openshift oauth server doesn't have a token verification endpoint, but I could talk to the auth team to make sure that's correct, and implement something here if we can.

Reply:
Okay, that makes sense. I looked a little more closely at [the JWT verifier](https://github.com/openshift/console/blob/1b1c0b8832b523f958e3a4a196677e3c227b8166/vendor/github.com/coreos/go-oidc/verify.go#L208) used by our OIDC authenticator and as you said, it doesn't look like there's anything similar we can do with an opaque token. It seems like we should consider decoupling this JWT-specific behavior from the sessions module at some point.